src/utils/database.ts

```diff
@@ -4417,46 +4585,78 @@ const createCryptoTransaction = async (
   if (error) {
     throw new Error(error.message)
   }
-  const meetingType = await getMeetingTypeFromDB(
-    transactionRequest.meeting_type_id
-  )
-  const totalNoOfSlots = meetingType?.plan?.no_of_slot || 1
-  const meetingSessions: Array<BaseMeetingSession> = Array.from(
-    { length: totalNoOfSlots },
-    (_, i) => ({
-      meeting_type_id: transactionRequest.meeting_type_id,
-      transaction_id: data[0]?.id,
-      session_number: i + 1,
-      guest_address: transactionRequest?.guest_address,
-      guest_email: transactionRequest?.guest_email,
-      owner_address: meetingType?.account_owner_address,
-    })
-  )
-  const { error: slotError } = await db.supabase
-    .from('meeting_sessions')
-    .insert(meetingSessions)
-  if (slotError) {
-    throw new Error(slotError.message)
-  }
-  try {
-    // don't wait for receipt to be sent before serving a response
-    sendReceiptEmail(
-      transactionRequest.guest_email,
-      transactionRequest.guest_name,
-      {
-        full_name: transactionRequest.guest_name,
-        email_address: transactionRequest.guest_email,
-        plan: meetingType.title,
-        number_of_sessions: totalNoOfSlots.toString(),
-        price: transactionRequest.amount.toString(),
-        payment_method: PaymentType.CRYPTO,
-        transaction_fee: '0',
-        transaction_status: PaymentStatus.COMPLETED,
-        transaction_hash: transactionRequest.transaction_hash,
+
+  // Only create meeting sessions and send receipt if this is a meeting-related transaction
+  if (transactionRequest.meeting_type_id) {
+    const meetingType = await getMeetingTypeFromDB(
+      transactionRequest.meeting_type_id
+    )
+    const totalNoOfSlots = meetingType?.plan?.no_of_slot || 1
+
+    // For meeting payments, add the meeting owner's address to metadata as receiver
+    if (meetingType?.account_owner_address) {
+      const { error: updateError } = await db.supabase
+        .from('transactions')
+        .update({
+          metadata: {
+            sender_address: account_address.toLowerCase(),
+            receiver_address: meetingType.account_owner_address.toLowerCase(),
```

Comment:
This is misleading; the receiver address could be a custom address, so doing this invalidates such actions. The transaction_hash should be the only source of truth